### PR TITLE
Fix unit tests for AGL

### PR DESCRIPTION
**Pull request by mkorniichuk**
This PR is **ready** for review.

### Summary
Most of the unit tests are failed with segfault on AGL. The issue is in gmock. The bug is already known and old enoug (https://bugs.launchpad.net/ubuntu/+source/google-mock/+bug/1609793).

NOTE: Current PR depends on https://github.com/Ford-Luxoft/sdl_core/pull/13

### Changelog
* Gmock segfault fix for AGL

### CLA
- [ ] I have signed [the CLA](https://docs.google.com/forms/d/e/1FAIpQLSdsgJY33VByaX482zHzi-xUm49JNnmuJOyAM6uegPQ2LXYVfA/viewform)

### CODE REVIEW

**Review by IGapchuk — APPROVED** [review]

File: tools/cmake/helpers/sources.cmake [hunk 1]
```diff
@@ -150,33 +150,29 @@ function(create_test NAME SOURCES LIBS)
...
-              ${CMAKE_SOURCE_DIR}/src/components/test_main.cc)
-  add_executable(
-    ${NAME}
-    EXCLUDE_FROM_ALL
+  add_executable("${NAME}"
+    ${CMAKE_SOURCE_DIR}/src/components/test_main.cc
     ${SOURCES}
   )
 
+  # TODO: Fix problems with Cotire on Windows and Qt APPLINK-28060
```
Comment (LuxoftAKutsan) [inline tools/cmake/helpers/sources.cmake]: @mkorniichuk  avoid links to internal jira in github

File: tools/cmake/helpers/sources.cmake [hunk 2]
```diff
@@ -150,33 +150,29 @@ function(create_test NAME SOURCES LIBS)
...
-    set(NAME "${NAME}_unity")
-  endif()
   target_link_libraries(${NAME} ${LIBS})
-  set_target_properties(
-    ${NAME}
-    PROPERTIES
-    EXCLUDE_FROM_ALL 0
-  )
   add_test(NAME ${NAME}
-  COMMAND ${NAME} --gtest_output=xml:${CMAKE_BINARY_DIR}/test_results/)
```
Comment (LuxoftAKutsan) [inline tools/cmake/helpers/sources.cmake:181]: Spaces, Check spaces

**Review by LuxoftAKutsan — CHANGES_REQUESTED** [review]

**Comment by LuxoftAKutsan** [comment]
Changes in gtest? Are you sure?

File: src/3rd_party-static/gmock-1.7.0/include/gmock/gmock-spec-builders.h [hunk 3]
```diff
@@ -1399,7 +1399,8 @@ class ActionResultHolder : public UntypedActionResultHolderBase {
 template <>
 class ActionResultHolder<void> : public UntypedActionResultHolderBase {
  public:
-  void GetValueAndDelete() const { delete this; }
+ explicit ActionResultHolder() {}
+   void GetValueAndDelete() const { delete this; }
```
Comment (ZhdanovP) [inline src/3rd_party-static/gmock-1.7.0/include/gmock/gmock-spec-builders.h]: @mkorniichuk Problems with indent level, isn't it?

**Review by ZhdanovP — APPROVED** [review]

**Review by LuxoftAKutsan — APPROVED** [review]

File: src/3rd_party-static/CMakeLists.GTest [hunk 4]
```diff
@@ -0,0 +1,15 @@
+cmake_minimum_required(VERSION 2.8.2)
```
Comment (BSolonenko) [inline src/3rd_party-static/CMakeLists.GTest:31]: @mkorniichuk may be 2.8.11?

Comment (BSolonenko) [reply src/3rd_party-static/CMakeLists.GTest:31]: @mkorniichuk  please add Copyright

**Review by BSolonenko — CHANGES_REQUESTED** [review]

File: src/3rd_party-static/CMakeLists.GTest (hunk 4 above)
Comment (BSolonenko) [reply src/3rd_party-static/CMakeLists.GTest:31]: @mkorniichuk I'm not sure we need this specifier in this file.

Comment (mkorniichuk) [reply src/3rd_party-static/CMakeLists.GTest:31]: @BSolonenko this cmake list will be triggered as a standalone project, so the **cmake_minimum_required** is a required statement. BTW, in such a way, we specify the minimum cmake version supported by gtest.

**Review by BSolonenko — APPROVED** [review]

**Pull request merged**